### ISSUE TITLE
Support Fine-grained Logging

### ISSUE THREAD

**Issue by haileyschoelkopf** [EleutherAI/lm-eval2#6]
We want users to be able to inspect the individual examples/predictions performed by their model.

This logging is done by the Promptsource/Bigscience fork of the lm-evaluation-harness: https://github.com/bigscience-workshop/lm-evaluation-harness . However, its actual implementation in the Bigscience version is quite hard to follow, so ideally we'd better support this inherently, by design.

**Comment by lintangsutawika** [comment]
Made a PR to try to support this.
